diff -up glib-2.22.5/gio/gunixmounts.c.bak glib-2.22.5/gio/gunixmounts.c
--- glib-2.22.5/gio/gunixmounts.c.bak	2009-04-01 01:04:20.000000000 +0200
+++ glib-2.22.5/gio/gunixmounts.c	2010-08-27 16:31:29.000000000 +0200
@@ -1944,6 +1944,10 @@ g_unix_mount_guess_should_display (GUnix
   mount_path = mount_entry->mount_path;
   if (mount_path != NULL)
     {
+      /* Hide mounts within a dot path, suppose it was a purpose to hide this mount */
+      if (g_strstr_len (mount_path, -1, "/.") != NULL)
+        return FALSE;
+
       if (g_str_has_prefix (mount_path, "/media/")) 
         {
           char *path;
